packages/datadog-plugin-fs/src/index.js

```diff
@@ -0,0 +1,490 @@
+'use strict'
+
+let kDirReadPromisified
+let kDirClosePromisified
+
+const tagMakers = {
+  open: createOpenTags,
+  close: createCloseTags,
+  readFile: createReadFileTags,
+  writeFile: createWriteFileTags,
+  appendFile: createAppendFileTags,
+  access: createPathTags,
+  copyFile: createCopyFileTags,
+  stat: createPathTags,
+  lstat: createPathTags,
+  fstat: createFDTags,
+  readdir: createPathTags,
+  opendir: createPathTags,
+  read: createFDTags,
+  write: createFDTags,
+  writev: createFDTags,
+  chmod: createChmodTags,
+  lchmod: createChmodTags,
+  fchmod: createFchmodTags,
+  chown: createChownTags,
+  lchown: createChownTags,
+  fchown: createFchownTags,
+  realpath: createPathTags,
+  readlink: createPathTags,
+  unlink: createPathTags,
+  symlink: createCopyFileTags,
+  link: createCopyFileTags,
+  rmdir: createPathTags,
+  rename: createCopyFileTags,
+  fsync: createFDTags,
+  fdatasync: createFDTags,
+  mkdir: createPathTags,
+  truncate: createPathTags,
+  ftruncate: createFDTags,
+  utimes: createPathTags,
+  futimes: createFDTags,
+  mkdtemp: createPathTags
+}
+
+const requiresParent = true
+
+function createWrapCreateReadStream (config, tracer) {
+  return function wrapCreateReadStream (createReadStream) {
+    return function wrappedCreateReadStream (path, options) {
+      if (!hasParent(tracer)) {
+        return createReadStream.apply(this, arguments)
+      }
+      const tags = makeFSFlagTags('ReadStream', path, options, 'r', config, tracer)
+      return tracer.trace('fs.operation', { tags }, (span, done) => {
+        const stream = createReadStream.apply(this, arguments)
+        stream.once('end', done)
+        stream.once('error', done)
+        return stream
+      })
+    }
+  }
+}
+
+function createWrapCreateWriteStream (config, tracer) {
+  return function wrapCreateWriteStream (createWriteStream) {
+    return function wrappedCreateWriteStream (path, options) {
+      if (!hasParent(tracer)) {
+        return createWriteStream.apply(this, arguments)
+      }
+      const tags = makeFSFlagTags('WriteStream', path, options, 'w', config, tracer)
+      return tracer.trace('fs.operation', { tags }, (span, done) => {
+        const stream = createWriteStream.apply(this, arguments)
+        stream.once('finish', done)
+        stream.once('error', done)
+        return stream
+      })
+    }
+  }
+}
+
+function createWrapExists (config, tracer) {
+  return function wrapExists (exists) {
+    return function wrappedExists (path, cb) {
```

Comment:
All the `wrappedX` functions should be renamed to `xWithTrace` to be consistent with other plugins. The idea behind the name is so that in the stack trace it's clear that the function was traced. With "wrapped" the user would know that it's a wrapper, but wouldn't know what's the wrapper.